mungegithub/mungers/approval-handler.go

```diff
@@ -90,8 +96,88 @@ func (h *ApprovalHandler) Munge(obj *github.MungeObject) {
 	}
 
 	approverSet := sets.String{}
+	if obj.Issue.User.Name != nil {
+		approverSet.Insert(*obj.Issue.User.Name)
+	}
+
+	approverSet = approverSet.Union(createApproverSet(comments))
+	needsApproval := h.getApprovalNeededFiles(files, approverSet)
+
+	if needsApproval.Len() > 0 {
+		glog.Infof("Canceling the approve label because these files %v have no valid approvers", needsApproval)
+		//if latest notification stale, update it
+		if notificationNeedsUpdate(obj) {
+			createMessage(obj, needsApproval)
+		}
+		if obj.HasLabel(approvedLabel) {
+			obj.RemoveLabel(approvedLabel)
+		}
+	} else if !obj.HasLabel(approvedLabel) {
+		obj.AddLabel(approvedLabel)
+	}
+
+}
+
+func notificationNeedsUpdate(obj *github.MungeObject) bool {
+	notificationMatcher := mungeComment.MungerNotificationName(approvalNotificationName)
+	comments, err := obj.ListComments()
+	if err != nil {
+		glog.Error("Could not list the comments for PR%v", obj.Issue.Number)
+		return false
+	}
+	latestNotification := mungeComment.FilterComments(comments, notificationMatcher).GetLast()
+	latestApprove := mungeComment.FilterComments(comments, mungeComment.CommandName("approve")).GetLast()
+	if latestNotification == nil {
+		return true
+	}
+	if latestApprove == nil {
+		// there was already a bot notitication and nothing has changed since
+		return false
+	}
+	if latestApprove.CreatedAt.After(*latestNotification.CreatedAt) {
+		// there has been approval since last notification
+		obj.DeleteComment(latestNotification)
+		return true
+	}
+	lastModified := obj.LastModifiedTime()
+	if latestApprove.CreatedAt.Before(*lastModified) {
+		obj.DeleteComment(latestNotification)
+		return true
+	}
+	return false
+}
 
-	// from oldest to latest
+func createMessage(obj *github.MungeObject, filesNeedApproval sets.String) {
+	files, err := obj.ListFiles()
```

Comment:
the name is wrong anyway. Can you return the message and post it from the calling function?